frappe/integrations/doctype/nextcloud_settings/nextcloud_settings.py

```diff
@@ -0,0 +1,207 @@
+
+		else:
+			if cint(self.backup_files):
+				db_backup, site_config, public_file_backup, private_file_backup = get_latest_backup_file(with_files=True)
+
+				if not public_file_backup or not private_file_backup:
+					generate_files_backup()
+					db_backup, site_config, public_file_backup, private_file_backup = get_latest_backup_file(with_files=True)
+			else:
+				db_backup, site_config,  = get_latest_backup_file()
+
+		return db_backup, site_config, public_file_backup, private_file_backup
+
+	def make_session(self):
+		session = requests.session()
+		session.verify = True
+		session.stream = True
+		session.auth = (self.email, self.get_password(fieldname='password',raise_exception=False))
+		session.headers.update({
+			"OCS-APIRequest": "true",
+		})
+		return session
+
+	def make_baseurl(self):
+		vurl = urlparse(self.nextcloud_url)
+		if not vurl.scheme:
+			return None
+		if not vurl.netloc:
+			return None
+		if not vurl.port:
+			port = 443 if vurl.scheme == 'https' else 80
+
+		base_url = '{0}://{1}:{2}'.format(vurl.scheme, vurl.netloc, vurl.port if vurl.port else port)
+		if self.webdav_url.startswith('/'):
+			base_url = '{0}{1}'.format(base_url, self.webdav_url)
+		else:
+			base_url = '{0}/{1}'.format(base_url, self.webdav_url)
+		if not base_url.endswith('/'):
+			base_url = '{0}/'.format(base_url)
+		return base_url
+
+	def check_for_upload_folder(self, session, url):
+		response = session.request("PROPFIND", url, headers={"Depth": "0"}, allow_redirects=False)
+		if response.status_code == 404:
+			if self.upload_path:
+				frappe.throw(_('Given "Path to upload folder" does not exist'))
+			else:
+				response = session.request("MKCOL", url, allow_redirects=False)
+				if response.status_code != 201:
+					frappe.throw(_('There was an error. Please try again'))
+
+def upload_backup(session, baseurl, filebackup):
+	if not os.path.exists(filebackup):
+		return
+	local_fileobj = filebackup
+	fileobj = local_fileobj.split('/')
+	dir_length = len(fileobj) - 1
+	remote_fileobj=fileobj[dir_length].encode("ascii", "ignore").decode("ascii")
+	if baseurl.endswith('/'):
+		url = '{0}{1}'.format(baseurl, remote_fileobj)
+	else:
+		url = '{0}/{1}'.format(baseurl, remote_fileobj)
+	if isinstance(filebackup, str):
+		try:
+			with open(filebackup, 'rb') as f:
+				response = session.request("PUT", url, allow_redirects=False, data=f)
+		except Exception as e:
+			return "Failed"
+	else:
+		try:
+			response = session.request("PUT", url, allow_redirects=False, data=filebackup)
+		except Exception as e:
+			return "Failed"
+	if response.status_code not in (201, 204):
+		return "Failed"
+	else:
+		return "Success"
+
+@frappe.whitelist()
+def take_backup():
+	"""Enqueue longjob for taking backup to nextcloud"""
+	enqueue("frappe.integrations.doctype.nextcloud_settings.nextcloud_settings.start_backup", queue='long', timeout=1500)
+	frappe.msgprint(_("Queued for backup. It may take a few minutes to an hour."))
```

Comment:
The 1500s timeout means 25m. So, it seems like at most, this will take 25 minutes. And if your upload doesn't happen in that timeframe, it gets killed, right?

So, at most, you can say this will take 30 minutes and not an hour